yt/odin/bin/yt_odin_webservice/__main__.py

```diff
@@ -151,6 +167,31 @@ def solomon_duration():
     return jsonify({"sensors": duration_sensors + timed_out_sensors})
 
 
+@app.route("/prom")
+def prometheus():
+    now = datetime.datetime.now(datetime.timezone.utc)
+    period = datetime.timedelta(seconds=int(request.args.get("period", 60)))
+    start_timestamp = int(calendar.timegm((now - period).utctimetuple()))
+    stop_timestamp = int(calendar.timegm(now.utctimetuple()))
+    db_records = extract_all_records(start_timestamp, stop_timestamp)
+
+    last_records_per_service = {}
+    for record in db_records:
+        if (
+                record.service not in last_records_per_service
+                or last_records_per_service[record.service].timestamp < record.timestamp
+        ):
+            last_records_per_service[record.service] = record
```

Comment:
I believe it's better to do this grouping on python backend side considering the small amount of data here.